README.md

```diff
@@ -20,41 +20,66 @@ Those familiar with the [`transformers`](https://github.com/huggingface/transfor
 
 Check out [our announcement post](https://huggingface.co/blog/swift-coreml-llm).
 
-## Modules
+## Examples
 
-- `Tokenizers`: Utilities to convert text to tokens and back, with support for Chat Templates and Tools. Follows the abstractions in [`tokenizers`](https://github.com/huggingface/tokenizers). 
+The most commonly used modules from `swift-transformers` are `Tokenizers` and `Hub`, which allow fast tokenization and
+model downloads from the Hugging Face Hub.
+
+### Tokenizing text + chat templating
+
+Tokenizing text should feel very familiar to those who have used the Python `transformers` library:
 
-Usage example:
 ```swift
-import Tokenizers
-func testTokenizer() async throws {
-    let tokenizer = try await AutoTokenizer.from(pretrained: "deepseek-ai/DeepSeek-R1-Distill-Qwen-7B")
-    let messages = [["role": "user", "content": "Describe the Swift programming language."]]
-    let encoded = try tokenizer.applyChatTemplate(messages: messages)
-    let decoded = tokenizer.decode(tokens: encoded)
-}
+let tokenizer = try await AutoTokenizer.from(pretrained: "deepseek-ai/DeepSeek-R1-Distill-Qwen-7B")
+let messages = [["role": "user", "content": "Describe the Swift programming language."]]
+let encoded = try tokenizer.applyChatTemplate(messages: messages)
+let decoded = tokenizer.decode(tokens: encoded)
 ```
-- `Hub`: Utilities for interacting with the Hugging Face Hub. Download models, tokenizers and other config files. 
 
-Usage example:
+
+### Tool calling
+
+`swift-transformers` natively supports tool calling, allowing for complex interactions with language models:
 
 ```swift
-import Hub
-func testHub() async throws {
-    let repo = Hub.Repo(id: "mlx-community/Qwen2.5-0.5B-Instruct-2bit-mlx")
-    let modelDirectory: URL = try await Hub.snapshot(
-        from: repo,
-        matching: ["config.json", "*.safetensors"],
-        progressHandler: { progress in
-            print("Download progress: \(progress.fractionCompleted * 100)%")
-        }
-    )
-    print("Files downloaded to: \(modelDirectory.path)")
-}
+let tokenizer = try await AutoTokenizer.from(pretrained: "mlx-community/Qwen2.5-7B-Instruct-4bit")
+
+let weatherTool = [
+    "type": "function",
+    "function": [
+        "name": "get_current_weather",
+        "description": "Get the current weather in a given location",
+        "parameters": [
+            "type": "object",
+            "properties": ["location": ["type": "string", "description": "City and state"]],
+            "required": ["location"]
+        ]
+    ]
+]
+
+let tokens = try tokenizer.applyChatTemplate(
+    messages: [["role": "user", "content": "What's the weather in Paris?"]],
+    tools: [weatherTool]
+)
```

Comment:
Perhaps we could write an additional, longer example using mlx for interaction with a language model, assuming there are tasks not fully undertaken by the mlx swift libraries.

Reply:
I think `mlx-swift-examples` only uses our tokenizers right? e.g https://github.com/ml-explore/mlx-swift-examples/blob/main/Libraries/MLXVLM/Models/Qwen25VL.swift#L733